src/languages/common.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{
+    lilypond_objects::lilypond_note::LilyPondNote,
+    notation::{
+        note::Note,
+        pitch::Octave,
+        rhythm::{DurationType, Length, Rhythm},
+    },
+};
+
+pub fn octave(note: &Note) -> &str {
+    match note.pitch.octave {
+        Octave::S0 => ",,,",
+        Octave::S1 => ",,",
+        Octave::S2 => ",",
+        Octave::S3 => "",
+        Octave::S4 => "'",
+        Octave::S5 => "''",
+        Octave::S6 => "'''",
+        Octave::S7 => "''''",
+        Octave::S8 => "'''''",
+        Octave::S9 => "''''''",
+        Octave::None => "",
+    }
+}
+
+pub fn length(note: &Note) -> &str {
+    match note.rhythm.length {
+        Length::Whole => "1",
+        Length::Half => "2",
+        Length::Quarter => "4",
+        Length::Eighth => "8",
+        Length::Sixteenth => "16",
+        Length::ThirtySecond => "32",
+        Length::SixtyFourth => "64",
+        Length::OneTwentyEighth => "128",
+    }
+}
```

Comment:
This could be done by having the value in the enum

```rust
#[repr(u16)]
pub enum Length {
    Whole = 1,
    Half = 2,
    Quarter = 4,
}

impl Length {
    pub fn as_u16(&self) -> u16 {
        self as u16
    }
}

impl Display for Length {
    fn fmt(&self, f &mut Formatter) -> Result<()> {
        write!(f, "{}", self.as_u16())
    }
}
```

The difference here is that we've implemented `to_string()` which returns a `String` and not a `&str`. For enums that can transform into `&str` instead of `String` I messed around with this https://github.com/webern/stringenum/blob/main/examples/main.rs

Also there's `serde_plain` for transforming enums to/from strings.

Anyway I'm just rambling about stuff that doesn't matter. Thank you for working on this stuff!

Reply:
I believe I have now done as you suggested; I have also replaced similar functions for dots and octaves with their own implementations of the `Display` trait.